src/polygonParts/models/polygonPartsManager.ts

```diff
@@ -198,48 +212,172 @@ export class PolygonPartsManager {
+      hasAnyGeometries,
+      requestFeatureId,
+    })
+      .from<FindPolygonPartsQueryResponse<ShouldClip>>(filterQuery.filterQueryAlias, filterQuery.filterQueryAlias)
+      .where(`st_geometrytype(${geometryColumn}) = :geometryType`, { geometryType: 'ST_Polygon' });
 
-    const clipCTE = polygonPart
-      .createQueryBuilder()
-      .select(canClip ? `(st_dump(st_intersection(${geometryColumn}, st_geomfromgeojson(:clipFootprint)))).geom` : geometryColumn, geometryColumn)
-      .addSelect(findSelectOutputColumns);
+    const findPolygonPartsQuery = this.buildPolygonPartsQueryWithFilter({
+      ...context,
+      filterQuery,
+      select: featureCollectionSelect,
+    });
+    return findPolygonPartsQuery;
+  }
 
-    const findPolygonPartsQuery = entityManager
-      .createQueryBuilder()
-      .select(
-        `jsonb_build_object(
+  private buildFindFeatureCollectionSelect(context: {
+    entityManager: EntityManager;
+    filterQuery: { filterQueryAlias: string; filterRequestFeatureIds: string };
+    geometryColumn: string;
+    hasAnyGeometries: boolean;
+    requestFeatureId: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  }): SelectQueryBuilder<any> {
+    const {
+      entityManager,
+      filterQuery: { filterQueryAlias, filterRequestFeatureIds },
+      geometryColumn,
+      hasAnyGeometries,
+      requestFeatureId,
+    } = context;
+    const requestFeatureIds = hasAnyGeometries
+      ? ` || jsonb_strip_nulls(
+      case
+        when array_length(${filterRequestFeatureIds}, 1) is null then '{}'::jsonb
+        when array_length(${filterRequestFeatureIds}, 1) = 1 then jsonb_build_object('${requestFeatureId}', ${filterRequestFeatureIds}[1])
+        else jsonb_build_object('${requestFeatureId}', ${filterRequestFeatureIds})
+      end
+    )`
+      : '';
+
+    const findPolygonPartsSelect = entityManager.createQueryBuilder().select(
+      `jsonb_build_object(
           'type', 'FeatureCollection',
           'features', coalesce(jsonb_agg(
             jsonb_build_object(
               'type', 'Feature',
               'geometry', st_asgeojson(${geometryColumn}, 15)::jsonb,
-              'properties', to_jsonb(clip) - '${geometryColumn}'
+              'properties', to_jsonb(${filterQueryAlias}) - '{${geometryColumn},${filterRequestFeatureIds}}'::text[] ${requestFeatureIds}
             )
           ), '[]')
         ) AS ${'geojson' satisfies keyof FindPolygonPartsQueryResponse}`
+    );
+    return findPolygonPartsSelect;
+  }
+
+  private buildPolygonPartsQueryWithFilter<T extends ObjectLiteral, ShouldClip extends boolean = boolean>(
+    context: FindPolygonPartsOptions<ShouldClip> & {
+      entityManager: EntityManager;
+      select: SelectQueryBuilder<T>;
+      filterQuery: { filterQueryAlias: string; filterRequestFeatureIds: string; findSelectOutputColumns: string[] };
+    }
+  ): SelectQueryBuilder<T> {
+    const {
+      entityManager,
+      filter,
```

Comment:
maybe we can merge 'filter' and 'filterQuery' as 'filterQueryOptions as one type and not seperated